Snackbar action color

<!-- Checklist for reporting a bug
- Check the issue tracker. The issue you have may have already been reported.
- Ensure you have tested the latest developers version.
-->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
The action text color in Snackbar is not consistent(for dark theme).

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
It should be black.

**Steps to reproduce the behavior:**
1. Open a ZIM
<!--
2. Click on Bookmark icon
3. See the action text(OPEN) color in snackbar
-->

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem otherwise delete this heading. -->
![photo_6140912444634152213_y](https://github.com/kiwix/kiwix-android/assets/85388413/51a97a94-f9bd-445d-8c29-950e8adcc29e)



**Environment**
- Version of Kiwix Android : 3.9.1
- Device : REALME GT
- OS version : 13

**Logs**
<!-- If applicable, add logs to help the developers in identifying your problem otherwise delete this heading. -->

Still not working, It's showing a lot of errors
```
Running lint...
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Configure project :app
[directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\install_time_asset\build']WARNING:BuildType(playStore): buildConfigField 'IS_PLAYSTORE' value is being replaced.
> Task :app:loadKtlintReporters FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:loadKtlintReporters'.
> Could not resolve all files for configuration ':app:ktlint'.
   > Could not resolve com.pinterest:ktlint:0.42.1.
     Required by:
         project :app
      > Could not resolve com.pinterest:ktlint:0.42.1.
         > Could not get resource 'https://dl.google.com/dl/android/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
            > Could not GET 'https://dl.google.com/dl/android/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
               > No such host is known (dl.google.com)
      > Could not resolve com.pinterest:ktlint:0.42.1.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
               > No such host is known (repo.maven.apache.org)
      > Could not resolve com.pinterest:ktlint:0.42.1.
         > Could not get resource 'https://jitpack.io/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
            > Could not GET 'https://jitpack.io/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
               > No such host is known (jitpack.io)
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 1m 17s
5 actionable tasks: 1 executed, 4 up-to-date
> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Configure project :app
[directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\install_time_asset\build']WARNING:BuildType(playStore): buildConfigField 'IS_PLAYSTORE' value is being replaced.
> Task :app:loadKtlintReporters FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:loadKtlintReporters'.
> Could not resolve all files for configuration ':app:ktlint'.
   > Could not resolve com.pinterest:ktlint:0.42.1.
     Required by:
         project :app
      > Could not resolve com.pinterest:ktlint:0.42.1.
         > Could not get resource 'https://dl.google.com/dl/android/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
            > Could not GET 'https://dl.google.com/dl/android/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
               > dl.google.com
      > Could not resolve com.pinterest:ktlint:0.42.1.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
               > repo.maven.apache.org
      > Could not resolve com.pinterest:ktlint:0.42.1.
         > Could not get resource 'https://jitpack.io/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
            > Could not GET 'https://jitpack.io/com/pinterest/ktlint/0.42.1/ktlint-0.42.1.pom'.
               > jitpack.io
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 3s
5 actionable tasks: 1 executed, 4 up-to-date
Static analysis found violations and attempted to autofix, please commit these autoformat changes
    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------
    If the build failed with '.../.gradle/daemon/5.6.1/custom/scr/customexample/info.json No File Found' then you do not have JAVA_HOME set to JDK8
    Please make sure JAVA_HOME is set to JDK8
    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------
21:23:49.295: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false rm --ignore-unmatch --cached -r -- core/src/main/res/values-b+be+tarask/strings.xml
21:26:19.084: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- .idea/inspectionProfiles/kiwixAndroidInspections.xml
21:26:26.897: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- core/src/main/res/values-b+be+tarask/strings.xml
21:26:31.820: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- core/src/main/res/values-b+be+tarask+old/strings.xml
21:26:44.691: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- core/objectbox-models/default.json.bak
21:26:49.408: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- core/objectbox-models/default.json
21:36:00.701: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- core/src/main/res/values-b+be+tarask+old/strings.xml
21:36:04.653: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false checkout HEAD -- core/src/main/res/values-b+be+tarask/strings.xml
21:52:37.017: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false rm --ignore-unmatch --cached -r -- core/src/main/res/values-b+be+tarask/strings.xml
21:52:37.119: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false add --ignore-errors -A -f -- core/src/main/res/values-b+be+tarask+old/strings.xml core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
warning: LF will be replaced by CRLF in core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt.
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt.
The file will have its original line endings in your working directory
21:52:38.033: [kiwix-android] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false commit -F C:\Users\sagar\AppData\Local\Temp\git-commit-msg-.txt --
Running lint...
> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Configure project :app
[directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\install_time_asset\build']WARNING:BuildType(playStore): buildConfigField 'IS_PLAYSTORE' value is being replaced.
> Task :app:loadKtlintReporters UP-TO-DATE
> Task :app:runKtlintCheckOverAndroidTestDebugSourceSet NO-SOURCE
> Task :app:ktlintAndroidTestDebugSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverAndroidTestNightlySourceSet NO-SOURCE
> Task :app:ktlintAndroidTestNightlySourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverAndroidTestPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintAndroidTestPlayStoreSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :app:ktlintAndroidTestReleaseSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverDebugSourceSet NO-SOURCE
> Task :app:ktlintDebugSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverNightlySourceSet NO-SOURCE
> Task :app:ktlintNightlySourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintPlayStoreSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverReleaseSourceSet NO-SOURCE
> Task :app:ktlintReleaseSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestDebugSourceSet NO-SOURCE
> Task :app:ktlintTestDebugSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestFixturesDebugSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesDebugSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestFixturesNightlySourceSet NO-SOURCE
> Task :app:ktlintTestFixturesNightlySourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestFixturesPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesPlayStoreSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestFixturesReleaseSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesReleaseSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestFixturesSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestNightlySourceSet NO-SOURCE
> Task :app:ktlintTestNightlySourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintTestPlayStoreSourceSetCheck SKIPPED
> Task :app:runKtlintCheckOverTestReleaseSourceSet NO-SOURCE
> Task :app:ktlintTestReleaseSourceSetCheck SKIPPED
> Task :core:loadKtlintReporters UP-TO-DATE
> Task :core:runKtlintCheckOverAndroidTestDebugSourceSet NO-SOURCE
> Task :core:ktlintAndroidTestDebugSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :core:ktlintAndroidTestReleaseSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverAndroidTestSourceSet NO-SOURCE
> Task :core:ktlintAndroidTestSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverDebugSourceSet NO-SOURCE
> Task :core:ktlintDebugSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverKotlinScripts
> Task :app:runKtlintCheckOverTestSourceSet
> Task :app:runKtlintCheckOverKotlinScripts
> Task :app:runKtlintCheckOverAndroidTestSourceSet
> Task :app:runKtlintCheckOverMainSourceSet
> Task :core:runKtlintCheckOverMainSourceSet
> Task :app:ktlintTestSourceSetCheck
> Task :core:runKtlintCheckOverReleaseSourceSet NO-SOURCE
> Task :core:ktlintReleaseSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverTestDebugSourceSet NO-SOURCE
> Task :core:ktlintTestDebugSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverTestFixturesDebugSourceSet NO-SOURCE
> Task :core:ktlintTestFixturesDebugSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverTestFixturesReleaseSourceSet NO-SOURCE
> Task :core:ktlintTestFixturesReleaseSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverTestFixturesSourceSet NO-SOURCE
> Task :core:ktlintTestFixturesSourceSetCheck SKIPPED
> Task :core:runKtlintCheckOverTestReleaseSourceSet NO-SOURCE
> Task :core:ktlintTestReleaseSourceSetCheck SKIPPED
> Task :app:ktlintKotlinScriptCheck
> Task :core:ktlintKotlinScriptCheck
> Task :custom:loadKtlintReporters UP-TO-DATE
> Task :custom:runKtlintCheckOverAndroidTestCustomexampleDebugSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestCustomexampleDebugSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverAndroidTestCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestCustomexampleSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverAndroidTestDebugSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestDebugSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestReleaseSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverAndroidTestSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverCustomexampleDebugSourceSet NO-SOURCE
> Task :custom:ktlintCustomexampleDebugSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverCustomexampleReleaseSourceSet NO-SOURCE
> Task :custom:ktlintCustomexampleReleaseSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintCustomexampleSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverDebugSourceSet NO-SOURCE
> Task :custom:ktlintDebugSourceSetCheck SKIPPED
> Task :app:ktlintAndroidTestSourceSetCheck
> Task :custom:runKtlintCheckOverReleaseSourceSet NO-SOURCE
> Task :custom:ktlintReleaseSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestCustomexampleDebugSourceSet NO-SOURCE
> Task :custom:ktlintTestCustomexampleDebugSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestCustomexampleReleaseSourceSet NO-SOURCE
> Task :custom:ktlintTestCustomexampleReleaseSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintTestCustomexampleSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestDebugSourceSet NO-SOURCE
> Task :custom:ktlintTestDebugSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestFixturesCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesCustomexampleSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestFixturesDebugSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesDebugSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestFixturesReleaseSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesReleaseSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestFixturesSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesSourceSetCheck SKIPPED
> Task :custom:runKtlintCheckOverTestReleaseSourceSet NO-SOURCE
> Task :custom:ktlintTestReleaseSourceSetCheck SKIPPED
> Task :app:ktlintMainSourceSetCheck
> Task :app:ktlintCheck
> Task :app:copyGitHooks UP-TO-DATE
> Task :app:installGitHooks SKIPPED
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :core:copyGitHooks UP-TO-DATE
> Task :core:installGitHooks SKIPPED
> Task :core:objectboxPrepareBuild UP-TO-DATE
> Task :core:preBuild UP-TO-DATE
> Task :core:preDebugBuild UP-TO-DATE
> Task :core:compileDebugAidl NO-SOURCE
> Task :app:compileDebugAidl NO-SOURCE
> Task :core:packageDebugRenderscript NO-SOURCE
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :custom:runKtlintCheckOverTestSourceSet
> Task :custom:runKtlintCheckOverMainSourceSet
> Task :custom:runKtlintCheckOverKotlinScripts
> Task :core:runKtlintCheckOverTestSourceSet
> Task :app:dataBindingMergeDependencyArtifactsDebug UP-TO-DATE
> Task :custom:ktlintKotlinScriptCheck
> Task :core:ktlintTestSourceSetCheck
> Task :core:ktlintMainSourceSetCheck
> Task :custom:ktlintMainSourceSetCheck
> Task :custom:ktlintTestSourceSetCheck
> Task :app:generateDebugResValues UP-TO-DATE
> Task :core:ktlintCheck
> Task :custom:ktlintCheck
> Task :app:generateDebugResources UP-TO-DATE
> Task :core:compileDebugRenderscript NO-SOURCE
> Task :core:generateDebugResValues UP-TO-DATE
> Task :core:generateDebugResources UP-TO-DATE
> Task :core:packageDebugResources
> Task :core:dataBindingMergeDependencyArtifactsDebug UP-TO-DATE
> Task :core:dataBindingGenBaseClassesDebug UP-TO-DATE
> Task :app:generateDebugBuildConfig UP-TO-DATE
> Task :app:generateSafeArgsDebug UP-TO-DATE
> Task :core:writeDebugAarMetadata UP-TO-DATE
> Task :app:mergeDebugResources
> Task :app:checkDebugAarMetadata UP-TO-DATE
> Task :app:dataBindingGenBaseClassesDebug UP-TO-DATE
> Task :app:mapDebugSourceSetPaths
> Task :app:createDebugCompatibleScreenManifests
> Task :app:extractDeepLinksDebug UP-TO-DATE
> Task :core:extractDeepLinksDebug UP-TO-DATE
> Task :core:processDebugManifest UP-TO-DATE
> Task :app:processDebugMainManifest
package="org.kiwix.kiwixmobile" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\src\main\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
> Task :app:processDebugManifest
package="org.kiwix.kiwixmobile" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build\intermediates\merged_manifest\debug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
package="org.kiwix.kiwixmobile" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build\intermediates\merged_manifest\debug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
package="org.kiwix.kiwixmobile" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build\intermediates\merged_manifest\debug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
package="org.kiwix.kiwixmobile" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build\intermediates\merged_manifest\debug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
> Task :core:compileDebugLibraryResources
> Task :core:generateDebugBuildConfig UP-TO-DATE
> Task :core:generateSafeArgsDebug UP-TO-DATE
> Task :core:parseDebugLocalResources
> Task :app:processDebugManifestForPackage
> Task :core:javaPreCompileDebug
> Task :core:javaPreCompileDebug UP-TO-DATE
> Task :core:generateDebugRFile UP-TO-DATE
> Task :core:generateDebugR2 UP-TO-DATE
> Task :custom:copyGitHooks
> Task :custom:installGitHooks SKIPPED
> Task :custom:preBuild UP-TO-DATE
> Task :custom:preCustomexampleDebugBuild UP-TO-DATE
> Task :custom:compileCustomexampleDebugAidl NO-SOURCE
> Task :custom:compileCustomexampleDebugRenderscript NO-SOURCE
> Task :custom:dataBindingMergeDependencyArtifactsCustomexampleDebug
> Task :custom:generateCustomexampleDebugResValues
> Task :custom:generateCustomexampleDebugResources
> Task :core:kaptGenerateStubsDebugKotlin
> Task :custom:generateCustomexampleDebugBuildConfig
> Task :custom:mergeCustomexampleDebugResources
> Task :custom:generateSafeArgsCustomexampleDebug
> Task :custom:mapCustomexampleDebugSourceSetPaths
> Task :custom:createCustomexampleDebugCompatibleScreenManifests
> Task :custom:checkCustomexampleDebugAarMetadata
> Task :custom:extractDeepLinksCustomexampleDebug
> Task :custom:processCustomexampleDebugMainManifest
package="org.kiwix.kiwixmobile.custom" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\src\main\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
> Task :custom:processCustomexampleDebugManifest
package="org.kiwix.kiwixcustomcustomexample" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build\intermediates\merged_manifest\customexampleDebug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
package="org.kiwix.kiwixcustomcustomexample" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build\intermediates\merged_manifest\customexampleDebug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
package="org.kiwix.kiwixcustomcustomexample" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build\intermediates\merged_manifest\customexampleDebug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
package="org.kiwix.kiwixcustomcustomexample" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build\intermediates\merged_manifest\customexampleDebug\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
> Task :core:mergeDebugJniLibFolders UP-TO-DATE
> Task :core:mergeDebugNativeLibs NO-SOURCE
> Task :core:stripDebugDebugSymbols NO-SOURCE
> Task :custom:dataBindingGenBaseClassesCustomexampleDebug
> Task :core:extractDeepLinksForAarDebug
> Task :core:mergeDebugShaders UP-TO-DATE
> Task :core:copyDebugJniLibsProjectAndLocalJars
> Task :core:compileDebugShaders NO-SOURCE
> Task :core:generateDebugAssets UP-TO-DATE
> Task :core:packageDebugAssets UP-TO-DATE
> Task :core:prepareDebugArtProfile
> Task :core:generateDebugJacocoPropertiesFile UP-TO-DATE
> Task :core:prepareLintJarForPublish
> Task :core:processDebugJavaRes NO-SOURCE
> Task :app:javaPreCompileDebug UP-TO-DATE
> Task :app:preDebugAndroidTestBuild SKIPPED
> Task :app:processDebugAndroidTestManifest
package="org.kiwix.kiwixmobile" found in source AndroidManifest.xml: C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\src\androidTest\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
> Task :custom:processCustomexampleDebugManifestForPackage
> Task :core:kaptDebugKotlin
> Task :core:kaptDebugKotlin UP-TO-DATE
> Task :app:compileDebugAndroidTestRenderscript NO-SOURCE
> Task :app:extractProguardFiles
> Task :app:generateDebugAndroidTestResValues
> Task :core:writeDebugLintModelMetadata
> Task :core:preDebugAndroidTestBuild UP-TO-DATE
> Task :core:processDebugAndroidTestManifest
> Task :core:compileDebugAndroidTestRenderscript NO-SOURCE
> Task :core:extractProguardFiles
> Task :core:generateDebugAndroidTestResValues
> Task :custom:javaPreCompileCustomexampleDebug
> Task :custom:preCustomexampleDebugAndroidTestBuild SKIPPED
> Task :custom:processCustomexampleDebugAndroidTestManifest
> Task :custom:compileCustomexampleDebugAndroidTestRenderscript NO-SOURCE
> Task :custom:extractProguardFiles
> Task :custom:generateCustomexampleDebugAndroidTestResValues
> Task :app:processDebugResources
> Task :custom:processCustomexampleDebugResources
> Task :core:compileDebugKotlin
w: C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\src\main\java\org\kiwix\kiwixmobile\core\main\CoreReaderFragment.kt: (381, 5): The corresponding parameter in the supertype 'FragmentActivityExtensions' is named 'actionMode'. This may cause problems when calling this function with named arguments.
w: C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\src\main\java\org\kiwix\kiwixmobile\core\main\CoreReaderFragment.kt: (382, 5): The corresponding parameter in the supertype 'FragmentActivityExtensions' is named 'activity'. This may cause problems when calling this function with named arguments.
w: C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\src\main\java\org\kiwix\kiwixmobile\core\main\CoreReaderFragment.kt: (1553, 28): Name shadowed: zimFileReader
w: C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\src\main\java\org\kiwix\kiwixmobile\core\main\CoreReaderFragment.kt: (2026, 37): The corresponding parameter in the supertype 'WebViewCallback' is named 'failingUrl'. This may cause problems when calling this function with named arguments.
> Task :core:compileDebugJavaWithJavac UP-TO-DATE
> Task :core:bundleLibCompileToJarDebug
> Task :core:detektDebug
The BindingContext was created with 312 issues. Run detekt with --debug to see the error messages.
> Task :app:kaptGenerateStubsDebugKotlin
> Task :custom:kaptGenerateStubsCustomexampleDebugKotlin
> Task :core:extractDebugAnnotations
> Task :core:mergeDebugGeneratedProguardFiles
> Task :core:mergeDebugConsumerProguardFiles
> Task :core:mergeDebugJavaResource
> Task :app:kaptDebugKotlin
> Task :custom:kaptCustomexampleDebugKotlin
> Task :core:transformDebugClassesWithAsm
> Task :app:compileDebugKotlin
> Task :custom:compileCustomexampleDebugKotlin
Compilation with Kotlin compile daemon was not successful
java.rmi.UnmarshalException: Error unmarshaling return header; nested exception is: 
	java.net.SocketException: Connection reset
	at java.rmi/sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:254)
	at java.rmi/sun.rmi.server.UnicastRef.invoke(UnicastRef.java:164)
	at java.rmi/java.rmi.server.RemoteObjectInvocationHandler.invokeRemoteMethod(RemoteObjectInvocationHandler.java:217)
	at java.rmi/java.rmi.server.RemoteObjectInvocationHandler.invoke(RemoteObjectInvocationHandler.java:162)
	at com.sun.proxy.$Proxy149.compile(Unknown Source)
	at org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork.incrementalCompilationWithDaemon(GradleKotlinCompilerWork.kt:298)
	at org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork.compileWithDaemon(GradleKotlinCompilerWork.kt:215)
	at org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork.compileWithDaemonOrFallbackImpl(GradleKotlinCompilerWork.kt:152)
	at org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork.run(GradleKotlinCompilerWork.kt:126)
	at org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction.execute(GradleCompilerRunnerWithWorkers.kt:79)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$0(DefaultWorkerExecutor.java:169)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:187)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.access$700(DefaultConditionalExecutionQueue.java:120)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner$1.run(DefaultConditionalExecutionQueue.java:162)
	at org.gradle.internal.Factories$1.create(Factories.java:31)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:249)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:109)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:114)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:157)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:126)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.SocketException: Connection reset
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:186)
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
	at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:271)
	at java.base/java.io.DataInputStream.readByte(DataInputStream.java:270)
	at java.rmi/sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:240)
	... 44 more
Unable to clear jar cache after compilation, maybe daemon is already down: java.rmi.ConnectException: Connection refused to host: 127.0.0.1; nested exception is: 
	java.net.ConnectException: Connection refused: connect
Could not connect to kotlin daemon. Using fallback strategy.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jetbrains.kotlin.kapt3.util.ModuleManipulationUtilsKt (file:/C:/Users/sagar/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-annotation-processing-gradle/1.7.0/4af73d85cbf3357a2580f883cc2446c6fb813d3f/kotlin-annotation-processing-gradle-1.7.0.jar) to constructor com.sun.tools.javac.util.Context()
WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.kapt3.util.ModuleManipulationUtilsKt
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
> Task :app:detektDebug
The BindingContext was created with 189 issues. Run detekt with --debug to see the error messages.
> Task :core:syncDebugLibJars
> Task :custom:compileCustomexampleDebugKotlin
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (malloc) failed to allocate 1584496 bytes for Chunk::new
# An error report file with more information is saved as:
# C:\Users\sagar\AndroidStudioProjects\kiwix-android\hs_err_pid24080.log
#
# Compiler replay data is saved as:
# C:\Users\sagar\AndroidStudioProjects\kiwix-android\replay_pid24080.log
> Task :custom:compileCustomexampleDebugKotlin FAILED
> Task :core:bundleDebugLocalLintAar
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':custom:compileCustomexampleDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 9m 32s
113 actionable tasks: 78 executed, 35 up-to-date
> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Configure project :app
[directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\app\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\core\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\custom\build', directory 'C:\Users\sagar\AndroidStudioProjects\kiwix-android\install_time_asset\build']WARNING:BuildType(playStore): buildConfigField 'IS_PLAYSTORE' value is being replaced.
> Task :app:loadKtlintReporters UP-TO-DATE
> Task :app:runKtlintFormatOverKotlinScripts UP-TO-DATE
> Task :app:runKtlintFormatOverAndroidTestDebugSourceSet NO-SOURCE
> Task :app:ktlintAndroidTestDebugSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverAndroidTestNightlySourceSet NO-SOURCE
> Task :app:ktlintAndroidTestNightlySourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverAndroidTestPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintAndroidTestPlayStoreSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :app:ktlintAndroidTestReleaseSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverAndroidTestSourceSet UP-TO-DATE
> Task :app:ktlintAndroidTestSourceSetFormat UP-TO-DATE
> Task :app:runKtlintFormatOverDebugSourceSet NO-SOURCE
> Task :app:ktlintDebugSourceSetFormat SKIPPED
> Task :app:ktlintKotlinScriptFormat UP-TO-DATE
> Task :app:runKtlintFormatOverMainSourceSet UP-TO-DATE
> Task :app:ktlintMainSourceSetFormat UP-TO-DATE
> Task :app:runKtlintFormatOverNightlySourceSet NO-SOURCE
> Task :app:ktlintNightlySourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintPlayStoreSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverReleaseSourceSet NO-SOURCE
> Task :app:ktlintReleaseSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestDebugSourceSet NO-SOURCE
> Task :app:ktlintTestDebugSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestFixturesDebugSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesDebugSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestFixturesNightlySourceSet NO-SOURCE
> Task :app:ktlintTestFixturesNightlySourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestFixturesPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesPlayStoreSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestFixturesReleaseSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesReleaseSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestFixturesSourceSet NO-SOURCE
> Task :app:ktlintTestFixturesSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestNightlySourceSet NO-SOURCE
> Task :app:ktlintTestNightlySourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestPlayStoreSourceSet NO-SOURCE
> Task :app:ktlintTestPlayStoreSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestReleaseSourceSet NO-SOURCE
> Task :app:ktlintTestReleaseSourceSetFormat SKIPPED
> Task :app:runKtlintFormatOverTestSourceSet UP-TO-DATE
> Task :app:ktlintTestSourceSetFormat UP-TO-DATE
> Task :app:ktlintFormat UP-TO-DATE
> Task :core:loadKtlintReporters UP-TO-DATE
> Task :core:runKtlintFormatOverKotlinScripts UP-TO-DATE
> Task :core:runKtlintFormatOverAndroidTestDebugSourceSet NO-SOURCE
> Task :core:ktlintAndroidTestDebugSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :core:ktlintAndroidTestReleaseSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverAndroidTestSourceSet NO-SOURCE
> Task :core:ktlintAndroidTestSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverDebugSourceSet NO-SOURCE
> Task :core:ktlintDebugSourceSetFormat SKIPPED
> Task :core:ktlintKotlinScriptFormat UP-TO-DATE
> Task :core:runKtlintFormatOverReleaseSourceSet NO-SOURCE
> Task :core:ktlintReleaseSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverTestDebugSourceSet NO-SOURCE
> Task :core:ktlintTestDebugSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverTestFixturesDebugSourceSet NO-SOURCE
> Task :core:ktlintTestFixturesDebugSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverTestFixturesReleaseSourceSet NO-SOURCE
> Task :core:ktlintTestFixturesReleaseSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverTestFixturesSourceSet NO-SOURCE
> Task :core:ktlintTestFixturesSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverTestReleaseSourceSet NO-SOURCE
> Task :core:ktlintTestReleaseSourceSetFormat SKIPPED
> Task :core:runKtlintFormatOverTestSourceSet UP-TO-DATE
> Task :core:ktlintTestSourceSetFormat UP-TO-DATE
> Task :custom:loadKtlintReporters UP-TO-DATE
> Task :custom:runKtlintFormatOverKotlinScripts UP-TO-DATE
> Task :custom:runKtlintFormatOverAndroidTestCustomexampleDebugSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestCustomexampleDebugSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverAndroidTestCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestCustomexampleSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverAndroidTestDebugSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestDebugSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestReleaseSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverAndroidTestSourceSet NO-SOURCE
> Task :custom:ktlintAndroidTestSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverCustomexampleDebugSourceSet NO-SOURCE
> Task :custom:ktlintCustomexampleDebugSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverCustomexampleReleaseSourceSet NO-SOURCE
> Task :custom:ktlintCustomexampleReleaseSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintCustomexampleSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverDebugSourceSet NO-SOURCE
> Task :custom:ktlintDebugSourceSetFormat SKIPPED
> Task :custom:ktlintKotlinScriptFormat UP-TO-DATE
> Task :custom:runKtlintFormatOverMainSourceSet UP-TO-DATE
> Task :custom:ktlintMainSourceSetFormat UP-TO-DATE
> Task :custom:runKtlintFormatOverReleaseSourceSet NO-SOURCE
> Task :custom:ktlintReleaseSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestCustomexampleDebugSourceSet NO-SOURCE
> Task :custom:ktlintTestCustomexampleDebugSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestCustomexampleReleaseSourceSet NO-SOURCE
> Task :custom:ktlintTestCustomexampleReleaseSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintTestCustomexampleSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestDebugSourceSet NO-SOURCE
> Task :custom:ktlintTestDebugSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestFixturesCustomexampleSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesCustomexampleSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestFixturesDebugSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesDebugSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestFixturesReleaseSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesReleaseSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestFixturesSourceSet NO-SOURCE
> Task :custom:ktlintTestFixturesSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestReleaseSourceSet NO-SOURCE
> Task :custom:ktlintTestReleaseSourceSetFormat SKIPPED
> Task :custom:runKtlintFormatOverTestSourceSet UP-TO-DATE
> Task :custom:ktlintTestSourceSetFormat UP-TO-DATE
> Task :custom:ktlintFormat UP-TO-DATE
> Task :core:runKtlintFormatOverMainSourceSet
> Task :core:ktlintMainSourceSetFormat
> Task :core:ktlintFormat
BUILD SUCCESSFUL in 20s
27 actionable tasks: 2 executed, 25 up-to-date
Static analysis found violations and attempted to autofix, please commit these autoformat changes
    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------
    If the build failed with '.../.gradle/daemon/5.6.1/custom/scr/customexample/info.json No File Found' then you do not have JAVA_HOME set to JDK8
    Please make sure JAVA_HOME is set to JDK8
    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------

```

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt
@@ -28,6 +28,20 @@ import com.google.android.material.snackbar.Snackbar
 @SuppressLint("ShowToast")
 fun View.snack(
   stringId: Int,
+  actionStringId: Int? = null,
+  actionClick: (() -> Unit)? = null,
+) {
+  Snackbar.make(
+    this, stringId, Snackbar.LENGTH_LONG
+  ).apply {
+    actionStringId?.let { setAction(it) { actionClick?.invoke() } }
+    show()
+  }
+}
+
+@SuppressLint("ShowToast")
+fun View.snack(
+  stringId: Int,
   anchor: View? = null,
   actionStringId: Int? = null,
   actionClick: (() -> Unit)? = null,
@@ -69,8 +83,4 @@ fun View.snack(
       }
     })
   }.show()
-}
-
-fun View.snack(stringId: Int) {
-  snack(stringId, null, null)
 }

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt
@@ -28,30 +28,16 @@ import com.google.android.material.snackbar.Snackbar
 @SuppressLint("ShowToast")
 fun View.snack(
   stringId: Int,
-  actionStringId: Int? = null,
-  actionClick: (() -> Unit)? = null,
-) {
-  Snackbar.make(
-    this, stringId, Snackbar.LENGTH_LONG
-  ).apply {
-    actionStringId?.let { setAction(it) { actionClick?.invoke() } }
-    show()
-  }
-}
-
-@SuppressLint("ShowToast")
-fun View.snack(
-  stringId: Int,
   anchor: View? = null,
   actionStringId: Int? = null,
   actionClick: (() -> Unit)? = null,
-  @ColorInt actionTextColor: Int = Color.WHITE
+  @ColorInt actionTextColor: Int? = null
 ) {
   Snackbar.make(
     this, stringId, Snackbar.LENGTH_LONG
   ).apply {
     actionStringId?.let { setAction(it) { actionClick?.invoke() } }
-    setActionTextColor(actionTextColor)
+    actionTextColor?.let(::setActionTextColor)
     anchor?.let {
       anchorView = anchor
       addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
@@ -69,13 +55,13 @@ fun View.snack(
   anchor: View,
   actionStringId: Int? = null,
   actionClick: (() -> Unit)? = null,
-  @ColorInt actionTextColor: Int = Color.WHITE
+  @ColorInt actionTextColor: Int? = null
 ) {
   Snackbar.make(
     this, message, Snackbar.LENGTH_LONG
   ).apply {
     actionStringId?.let { setAction(it) { actionClick?.invoke() } }
-    setActionTextColor(actionTextColor)
+    actionTextColor?.let(::setActionTextColor)
     anchorView = anchor
     addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
       override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt
@@ -19,7 +19,6 @@
 package org.kiwix.kiwixmobile.core.extensions
 
 import android.annotation.SuppressLint
-import android.graphics.Color
 import android.view.View
 import androidx.annotation.ColorInt
 import com.google.android.material.snackbar.BaseTransientBottomBar

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
@@ -2101,21 +2101,15 @@ abstract class CoreReaderFragment :
       {
         if (isOpenNewTabInBackground) {
           newTabInBackground(url)
-          snackBarRoot?.let {
-            Snackbar.make(it, R.string.new_tab_snack_bar, Snackbar.LENGTH_LONG)
-              .setAction(getString(R.string.open)) {
-                if (webViewList.size > 1) selectTab(
-                  webViewList.size - 1
-                )
-              }
-              .setActionTextColor(
-                ContextCompat.getColor(
-                  requireActivity(),
-                  R.color.alabaster_white
-                )
+          snackBarRoot?.snack(
+            stringId = R.string.new_tab_snack_bar,
+            actionStringId = R.string.open,
+            actionClick = {
+              if (webViewList.size > 1) selectTab(
+                webViewList.size - 1
               )
-              .show()
-          }
+            }
+          )
         } else {
           newTab(url)
         }

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
@@ -1686,11 +1686,7 @@ abstract class CoreReaderFragment :
               actionClick = {
                 goToBookmarks()
                 Unit
-              },
-              actionTextColor = ContextCompat.getColor(
-                requireActivity(),
-                R.color.alabaster_white
-              )
+              }
             )
           }
         } ?: kotlin.run {

